pallets/valve/src/benchmarking.rs

```diff
@@ -19,45 +19,81 @@
 
 use super::*;
 use frame_benchmarking::benchmarks;
+use frame_support::traits::SortedMembers;
 use frame_system::RawOrigin;
 
+fn assert_last_event<T: Config>(generic_event: <T as Config>::Event) {
+	frame_system::Pallet::<T>::assert_last_event(generic_event.into());
+}
+
+fn get_caller_account_id<T: Config>() -> T::AccountId {
+	T::CallAccessFilter::sorted_members().pop().unwrap()
+}
+
 benchmarks! {
 	close_valve {
-
-	}: _(RawOrigin::Root)
+		let caller = get_caller_account_id::<T>();
+	}: _(RawOrigin::Signed(caller))
+	verify {
+		assert_last_event::<T>(Event::ValveClosed.into())
+	}
 
 	open_valve {
+		let caller = get_caller_account_id::<T>();
 		ValveClosed::<T>::put(true);
-	}: _(RawOrigin::Root)
+	}: _(RawOrigin::Signed(caller))
+	verify {
+		assert_last_event::<T>(Event::ValveOpen.into())
+	}
 
 	close_pallet_gate_new {
-
-	}: close_pallet_gate(RawOrigin::Root, b"System".to_vec())
+		let caller = get_caller_account_id::<T>();
+		let pallet_name = b"System".to_vec();
+	}: close_pallet_gate(RawOrigin::Signed(caller), pallet_name.clone())
+	verify {
+		assert_last_event::<T>(Event::PalletGateClosed{ pallet_name_bytes: pallet_name }.into())
+	}
 
 	close_pallet_gate_existing {
-		ClosedPallets::<T>::insert(b"System".to_vec(), ());
-	}: close_pallet_gate(RawOrigin::Root, b"System".to_vec())
-
+		let caller = get_caller_account_id::<T>();
+		let pallet_name = b"System".to_vec();
+		ClosedPallets::<T>::insert(pallet_name.clone(), ());
+	}: close_pallet_gate(RawOrigin::Signed(caller), pallet_name.to_vec())
 
 	open_pallet_gate {
+		let caller = get_caller_account_id::<T>();
 		let pallet_name = b"System".to_vec();
 		ClosedPallets::<T>::insert(pallet_name.clone(), ());
-	}: _(RawOrigin::Root, pallet_name)
+	}: _(RawOrigin::Signed(caller), pallet_name.clone())
+	verify {
+		assert_last_event::<T>(Event::PalletGateOpen{ pallet_name_bytes: pallet_name }.into())
+	}
 
 	open_pallet_gates {
+		let caller = get_caller_account_id::<T>();
 		ClosedPallets::<T>::insert(b"System".to_vec(), ());
 		ClosedPallets::<T>::insert(b"Balances".to_vec(), ());
 		ClosedPallets::<T>::insert(b"Bounties".to_vec(), ());
 		ClosedPallets::<T>::insert(b"CollatorSelection".to_vec(), ());
 		ClosedPallets::<T>::insert(b"Treasury".to_vec(), ());
 		ClosedPalletCount::<T>::put(5);
-	}: _(RawOrigin::Root)
+	}: _(RawOrigin::Signed(caller))
+	verify {
+		assert_last_event::<T>(Event::PalletGatesClosed{ count: 0 }.into())
+	}
 
 	stop_scheduled_tasks {
-
-	}: _(RawOrigin::Root)
+		let caller = get_caller_account_id::<T>();
+	}: _(RawOrigin::Signed(caller))
+	verify {
+		assert_last_event::<T>(Event::ScheduledTasksStopped.into())
+	}
 
 	start_scheduled_tasks {
+		let caller = get_caller_account_id::<T>();
 		T::AutomationTime::shutdown();
-	}: _(RawOrigin::Root)
+	}: _(RawOrigin::Signed(caller))
+	verify {
```

Comment:
there was general weight increase across the board in the weights. was it because of the verify? or was it really just because there's an extra if in the logic?

Reply:
Verify does not add weight. It's just a test to the state after the benchmark is run